### PR TITLE
Create FHIR resource: ResourceCategory component

### PR DESCRIPTION
**Pull request by michal-maksajda**
Issue:  [Create FHIR resource: ResourceCategory component](https://app.zenhub.com/workspaces/patient-app-6140de7e31081800158d8ecf/issues/1uphealth/fhir-react/279) 

---

##### PR Checklist
- [x] Give this PR a meaningful title
- [x] Add a link to the related issue at the top of this description (above)
- [x] Connect this PR with the related issue via ZenHub with the button below this text box (or at the bottom of the page after the PR is created)
- [x] Ensure your branch is up to date with the target branch and resolve any conflicts
- [x] Answer the below questions to describe your PR for reviewers
- [x] Request at least two reviewers using the "Reviewers" section on the right, usually including at least one reviewer from your team
- [x] Notify the requested reviewers in the #code-review Slack channel once the PR is ready for review

---

## Why are these changes needed?

Due to the latest redesign, which introduced bootstrapped version of UI components, creating ResourceCategory was necessary to fulfill the requirements.

## What changed?

The ResourceCategory component has been created. Title & HeaderIcon components were used as the nested parts of the main component.
Zrzut ekranu 2021-11-10 o 11 16 07

## How are these changes tested?

- All automatic tests are passing
- The ResourceCategory component might be tested manually by viewing the current version of deployed [storybook](https://fhir-react-lib-test-storybook.s3.amazonaws.com/branch/add-FHIR-resource-resourcecategory/index.html?path=/story/resourcecategory--example-without-items-count)

### CODE REVIEW

File: src/components/resources/ResourceCategory/ResourceCategory.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          <Title data-testid="resourceCategoryName">{title}</Title>
+        </div>
+        <div className="d-flex gap-3">
+          {isPositiveNumber(itemsCount) && (
+            <div
+              className="rounded-pill py-1 px-2 bg-primary"
+              data-testid="itemsCount"
+            >
+              <span className="text-white fw-bold text-nowrap">
+                {itemsCount} {Number(itemsCount) === 1 ? 'item' : 'items'}
```
Comment (sflejszman) [inline src/components/resources/ResourceCategory/ResourceCategory.js:33]: What do you think about creating a function to pluralize words?

File: src/components/resources/ResourceCategory/ResourceCategory.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      <button
+        type="button"
+        className="btn d-flex align-items-center justify-content-between w-100 py-3 px-4"
+      >
+        <div className="d-flex gap-2">
+          <HeaderIcon headerIcon={headerIcon} />
+          <Title data-testid="resourceCategoryName">{title}</Title>
+        </div>
+        <div className="d-flex gap-3">
+          {isPositiveNumber(itemsCount) && (
```
Comment (sflejszman) [inline src/components/resources/ResourceCategory/ResourceCategory.js:27]:
```suggestion
          {itemsCount > 0 && (
```

Comment (sflejszman) [reply src/components/resources/ResourceCategory/ResourceCategory.js:27]: Of course, before we need to parse the string to number

File: src/assets/containers/ResourceCategory/chevron-right.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-chevron-right" viewBox="0 0 16 16">
```
Comment (sflejszman) [inline src/assets/containers/ResourceCategory/chevron-right.svg:1]: What do you think about creating the chevron icon as a common asset?

**Review by sflejszman — CHANGES_REQUESTED** [review]
Great work 🚀  I left some suggestions ;)

File: src/components/resources/ResourceCategory/ResourceCategory.test.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    const { container, getByTestId, getByAltText } = render(
+      <ResourceCategory {...defaultProps} />,
+    );
+    expect(container).not.toBeNull();
+
+    expect(getByTestId('itemsCount').textContent).toContain('1 item');
+    expect(getByTestId('title').textContent).toContain('Resource name');
+    const image = getByAltText(placeholderResource.alt);
+    expect(image.src).toContain(placeholderResource.src);
+  });
```
Comment (pslawinski-htd) [inline src/components/resources/ResourceCategory/ResourceCategory.test.js:44]: Let's also test against icorrect source data (`itemsCount: -1 // or 0 or 'asdf'`)

**Review by pslawinski-htd — CHANGES_REQUESTED** [review]

**Review by pslawinski-htd — APPROVED** [review]

**Review by sflejszman — APPROVED** [review]
🚀

**Comment by beck1uphealth** [comment]
Is anything supposed to happen when you click? When I try in storybook, nothing happens

**Comment by liam1uphealth** [comment]
@beck1uphealth it should link to https://fhir-react-lib-test-storybook.s3.amazonaws.com/branch/add-FHIR-resource-resourcecategory/index.html?path=/story/resourcecategory--example-without-items-count

**Comment by beck1uphealth** [comment]
@liam1uphealth Yeah, I am able to see the storybook, I was wondering if when you click on the resource component in the storybook if it should do anything since it is a button. I see the different ones that say 1 item and 41 items, but I was expecting it to do something when I clicked it

**Comment by michal-maksajda** [comment]
Hi. This ResourceCategory component is a template of the button. We should be able to pass items count as a property, for this reason, there are different variants listed in a storybook. Currently, the button has no onClick behaviour specified.

**Review by beck1uphealth — APPROVED** [review]

**Pull request merged**